gluten-substrait/src/main/scala/org/apache/spark/sql/execution/ColumnarBroadcastExchangeExec.scala

```diff
@@ -42,7 +42,8 @@ import scala.util.control.NonFatal
 
 case class ColumnarBroadcastExchangeExec(mode: BroadcastMode, child: SparkPlan)
   extends BroadcastExchangeLike
-  with ValidatablePlan {
+  with ValidatablePlan
+  with WithResourceProfileSupport {
```

Comment:
What if the exchange operators themselves fell back? Say a vanilla exchange with gluten plan as input.

Reply:
> Do we rely on WholeStageTransformer to set the resource profile in the case?
yes.
`WholeStageTransformer ` also implment `WithResourceProfileSupport` to have ability set ResourceProfile